modules/metastorage-server/src/main/java/org/apache/ignite/internal/metastorage/server/persistence/RocksDBKeyValueStorage.java

```diff
@@ -0,0 +1,1093 @@
+    private static final long SYSTEM_REVISION_MARKER_VALUE = -1;
+
+    /** Revision key. */
+    private static final byte[] REVISION_KEY = keyToRocksKey(
+        SYSTEM_REVISION_MARKER_VALUE,
+        "SYSTEM_REVISION_KEY".getBytes(StandardCharsets.UTF_8)
+    );
+
+    /** Update counter key. */
+    private static final byte[] UPDATE_COUNTER_KEY = keyToRocksKey(
+        SYSTEM_REVISION_MARKER_VALUE,
+        "SYSTEM_UPDATE_COUNTER_KEY".getBytes(StandardCharsets.UTF_8)
+    );
+
+    static {
+        RocksDB.loadLibrary();
+    }
+
+    /** RockDB options. */
+    private final DBOptions options;
+
+    /** RocksDb instance. */
+    private final RocksDB db;
+
+    /** Data column family. */
+    private final ColumnFamily data;
+
+    /** Index column family. */
+    private final ColumnFamily index;
+
+    /** RW lock. */
+    private final ReadWriteLock rwLock = new ReentrantReadWriteLock();
+
+    /** Thread-pool for snapshot operations execution. */
+    private final Executor snapshotExecutor = Executors.newFixedThreadPool(2);
+
+    /**
+     * Special value for the revision number which means that operation should be applied
+     * to the latest revision of an entry.
+     */
+    private static final long LATEST_REV = -1;
+
+    /** Lexicographic order comparator. */
+    static final Comparator<byte[]> CMP = Arrays::compare;
+
+    /** Path to the rocksdb database. */
+    private final Path dbPath;
+
+    /** Revision. Will be incremented for each single-entry or multi-entry update operation. */
+    private long rev;
+
+    /** Update counter. Will be incremented for each update of any particular entry. */
+    private long updCntr;
+
+    /**
+     * Constructor.
+     *
+     * @param dbPath RocksDB path.
+     */
+    public RocksDBKeyValueStorage(Path dbPath) {
+        try {
+            options = new DBOptions()
+                .setCreateMissingColumnFamilies(true)
+                .setCreateIfMissing(true);
+
+            this.dbPath = dbPath;
+
+            Options dataOptions = new Options().setCreateIfMissing(true)
+                // The prefix is the revision of an entry, so prefix length is the size of a long
+                .useFixedLengthPrefixExtractor(Long.BYTES);
+
+            ColumnFamilyOptions dataFamilyOptions = new ColumnFamilyOptions(dataOptions);
+
+            Options indexOptions = new Options().setCreateIfMissing(true);
+
+            ColumnFamilyOptions indexFamilyOptions = new ColumnFamilyOptions(indexOptions);
+
+            List<ColumnFamilyDescriptor> descriptors = Arrays.asList(
+                new ColumnFamilyDescriptor(DATA.nameAsBytes(), dataFamilyOptions),
+                new ColumnFamilyDescriptor(INDEX.nameAsBytes(), indexFamilyOptions)
+            );
+
+            var handles = new ArrayList<ColumnFamilyHandle>();
+
+            // Delete existing data, relying on the raft's snapshot and log playback
+            destroyRocksDB();
```

Comment:
are you sure this is needed?

Reply:
I think so, yes, we don't need a previous instance, Raft should handle everything